pkg/controller/controller.go

```diff
@@ -140,6 +140,8 @@ const (
 	snapshotNotBound = "snapshot %s not bound"
 
 	pvcCloneFinalizer = "provisioner.storage.kubernetes.io/cloning-protection"
+
+	annAllowVolumeModeChange = "snapshot.storage.kubernetes.io/allowVolumeModeChange"
```

Comment:
may be `allow-volume-mode-change` to be consistent with other similar constants? @RaunakShah

Reply:
@humblec I think we plan to change it in 1.25 but I don't mind doing that now too. 
@xing-yang what do you think?

Reply:
This annotation is already added to K8s 1.24 documentation. So we can't change it here now.  We can consider changing it in 1.25.